src/ORM/DataObject.php

```diff
@@ -1945,6 +1942,7 @@ public function setEagerLoadedData(
         string $eagerLoadRelation,
         EagerLoadedList|DataObject $eagerLoadedData
     ): void {
+        $this->objCacheClear();
```

Comment:
We forgot to add this when we implemented this method. Mirrors `setComponent()` and used to ensure stale data isn't returned from `obj()` or in the view layer.